modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3392,7 +3392,7 @@ public CodegenProperty fromProperty(String name, Schema p) {
         ModelUtils.syncValidationProperties(p, property);
 
         property.name = toVarName(name);
-        property.baseName = name;
+        property.baseName = escapeUnsafeCharacters(name);
```

Comment:
the baseName shouldn't be modified as it should stored the original `name` obtained from the spec.